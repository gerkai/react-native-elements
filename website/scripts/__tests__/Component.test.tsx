import 'react-native';
import { generateDocumentation } from '../docgen/generateDocumentation';
import path from 'path';
import fs from 'fs';

it('generates markdown correctly', async () => {
  const files = [path.join(__dirname, '..', 'Component.tsx')];
  await generateDocumentation(files);
  const filePresent = fs.existsSync(
    path.join(__dirname, '..', '..', 'docs', 'main', 'Component.md')
  );
  expect(filePresent).toBe(true);
});

it('validate generated markdown', async () => {
  const markdown = await fs.readFileSync(
    path.join(__dirname, '..', '..', 'docs', 'main', 'Component.md'),
    'utf8'
  );
  fs.unlinkSync(
    path.join(__dirname, '..', '..', 'docs', 'main', 'Component.md')
  );
  expect(markdown).toMatchSnapshot();
});
