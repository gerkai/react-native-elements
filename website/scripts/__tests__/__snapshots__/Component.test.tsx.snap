// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate generated markdown 1`] = `
"---


id: component


title: Component


slug: /component

---



import Usage from './usage/Component/Component.md'



The docs of the Component.



## Usage


<Usage />

---


## Props

### Component


 - [color](#color)
    
 - [type](#type)
    


## Reference

### Component
#### color


Color of the Component.

| Type | Default |
| ---- | ------- |
| string | demoColor |

---
#### type


Type of the Component.

| Type | Default |
| ---- | ------- |
| string | demoType |

---

"
`;
